dotnet/src/SemanticKernel.Abstractions/Search/TextSearchResult.cs

```diff
@@ -0,0 +1,48 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System.Diagnostics.CodeAnalysis;
+
+namespace Microsoft.SemanticKernel.Search;
+
+/// <summary>
+/// Represents a search result retrieved from a <see cref="ITextSearch" /> instance.
+/// </summary>
+/// <remarks>
+/// An instance of <see cref="TextSearchResult"/> is a normalised search result which provides access to:
+/// - Name associated with the search result
+/// - Value associated with the search result
+/// - Link reference associated with the search result
+/// </remarks>
+/// <param name="name">The text search result name.</param>
+/// <param name="value">The text search result value.</param>
+/// <param name="link">The link reference associated with the text search result.</param>
+[Experimental("SKEXP0001")]
+public sealed class TextSearchResult(string? name = null, string? value = null, string? link = null)
+{
+    /// <summary>
+    /// The text search result name.
+    /// </summary>
+    /// <remarks>
+    /// This represents the name associated with the result.
+    /// If the text search was for a web search engine this would typically be the name of the web page associated with the search result.
+    /// </remarks>
+    public string? Name { get; init; } = name;
+
+    /// <summary>
+    /// The link reference associated with the text search result.
+    /// </summary>
+    /// <remarks>
+    /// This represents a possible link associated with the result.
+    /// If the text search was for a web search engine this would typically be the URL of the web page associated with the search result.
+    /// </remarks>
+    public string? Link { get; init; } = link;
```

Comment:
nit: If URL of the web page will be a common case, I think `Link` name should work. If it can be URL or Path to file for example, word `Reference` may work a little bit better.

Reply:
Tracking here #8338